plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteInsertBatchImpl.java

```diff
@@ -159,54 +166,77 @@ StringBuilder prepareQueryForStatement(
             query.append(attributeName);
         }
         query.append(")\n\tVALUES "); //$NON-NLS-1$
+        
+        // Filtered attributes to remove columns that are ignored
+        DBSAttributeBase[] filteredAttributes = null;
+        DBDValueHandler[] filteredHandlers = null;
+        Object[] filteredAttributeValues = null;
+
+        if (skipAttributes.size() > 0) {
+            List<DBSAttributeBase> attributesList = new ArrayList<DBSAttributeBase>(Arrays.asList(attributes));
+            filteredAttributes = 
+                    IntStream.range(0, attributesList.size())
+                            .filter(i -> !skipAttributes.contains(i))
+                            .mapToObj(attributesList::get)
+                            .collect(Collectors.toList())
+                            .stream().toArray(DBSAttributeBase[]::new);
```

Comment:
See previous comment.

We can do following instead:
```suggestion
            filteredAttributes = usedAttributes.stream()
                .map(x -> attributes[x])
                .toArray(DBSAttributeBase[]::new);
```

Reply:
Stopped using filtered attributes